core/templates/pages/exploration-player-page/services/concept-card-manager.service.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility service for Concept Card in the learner's view.
+ */
+
+import { EventEmitter, Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { StateCard } from 'domain/state_card/state-card.model';
+
+import { ExplorationPlayerConstants } from 'pages/exploration-player-page/exploration-player-page.constants';
+import { PlayerPositionService } from 'pages/exploration-player-page/services/player-position.service';
+import { ExplorationEngineService } from './exploration-engine.service';
+import { HintsAndSolutionManagerService } from './hints-and-solution-manager.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ConceptCardManagerService {
+  // The following are set to null when the timeouts are cleared
+  // or when the service is reset.
+  timeout: NodeJS.Timeout | null = null;
+  tooltipTimeout: NodeJS.Timeout | null = null;
+  hintsAvailable: number = 0;
+
+  WAIT_FOR_TOOLTIP_TO_BE_SHOWN_MSEC: number = 2000;
+
+  // Emitter to show that the learner is really stuck.
+  _learnerReallyStuckEventEmitter = new EventEmitter();
+  
+  private _timeoutElapsedEventEmitter = new EventEmitter();
+  onTimeoutElapsed$ = this._timeoutElapsedEventEmitter.asObservable();
+
+  conceptCardReleased: boolean = false;
+  conceptCardConsumed: boolean = false;
+  wrongAnswersSinceConceptCardConsumed: number = 0;
+  learnerIsReallyStuck: boolean = false;
+
+
+  // Variable tooltipIsOpen is a flag which says that the tooltip is currently
+  // visible to the learner.
+  tooltipIsOpen: boolean = false;
+  // This is set to true as soon as the concept card icon is clicked or when
+  // the tooltip has been triggered.
+  conceptCardDiscovered: boolean = false;
+
+  constructor(
+    private playerPositionService: PlayerPositionService,
+    private hintsAndSolutionManagerService: HintsAndSolutionManagerService,
+    private explorationEngineService: ExplorationEngineService
+    ) {
+    // TODO(#10904): Refactor to move subscriptions into components.
```

Comment:
Why not just do it in this PR since it adds the service?

Reply:
I actually used this comment since I found that it has to be done as a future task after looking at other similar services. But there is actually no component for the concept-card icon separately, so it would be accommodated in the footer-component somewhere. Since the project deadline is close and am working on multiple PRs now and this has only one subscription, I thought it can be done later with the subscriptions.